modules/bloop-rifle/src/main/scala/scala/build/blooprifle/BloopRifle.scala

```diff
@@ -172,4 +184,51 @@ object BloopRifle {
     }
   }
 
+  def nullOutputStream() = new FileOutputStream(Util.devNull)
+
+  def nullInputStream() = new FileInputStream(Util.devNull)
+
+  def extractVersionFromBloopAbout(stdoutFromBloopAbout: String): Option[String] = {
+    stdoutFromBloopAbout.split("\n").lift(0).map(_.split(" ")(1).trim().drop(1))
+  }
+
+  def shutdownBloopIfVersionIncorrect(
+    config: BloopRifleConfig,
+    logger: BloopRifleLogger,
+    workdir: Path
+  ) {
+    val bufferedOStream = new ByteArrayOutputStream(100000)
+    val result = Operations.about(
+      config.host,
+      config.port,
+      workdir,
+      nullInputStream(),
+      bufferedOStream,
+      nullOutputStream(),
+      logger
+    )
+    if (result == 0) {
+      val currentBloopVersion =
+        extractVersionFromBloopAbout(new String(bufferedOStream.toByteArray())).map(Version(_))
+      val minimumBloopVersion = Version(Constants.bloopVersion)
```

Comment:
The minimum version should be put in `BloopRifleConfig`, with type `Option[String]` say. And no checks would be run if its value is `None`.

Reply:
Actually, we should just put a field like
```scala
acceptVersion: Option[String => Boolean]
```
accepting or discarding the Bloop version. That way, we wouldn't have to add new dependencies to the `bloop-rifle` module.

That field can default to `None`, and we can change that value around [here](https://github.com/VirtusLab/scala-cli/blob/130e0d763f9b146239304db6f35c6520fabc8a12/modules/cli/src/main/scala/scala/cli/commands/SharedCompilationServerOptions.scala#L164).

Reply:
For version comparison, coursier should be around there. I think something like this should work:
```scala
import coursier.core.Version
Version("1.2.3").compareTo(Version("1.4.5")) < 0
```

Reply:
Commented here https://github.com/VirtusLab/scala-cli/pull/145#discussion_r717327973